lib/ui/common/ActionSheet.dart

```diff
@@ -0,0 +1,200 @@
+  }
+
+  Widget build(BuildContext context) {
+    final Color cornersColor = const Color(0xFF737373);
+    return Container(
+        color: cornersColor,
+        child: Container(
+            decoration: BoxDecoration(
+                color: _style.backgroundColor,
+                borderRadius: BorderRadius.only(
+                    topLeft: _style.cornerRadius,
+                    topRight: _style.cornerRadius)),
+            child: Column(
+              children: <Widget>[
+                _buildIndicatorLine(_style),
+                Expanded(
+                  child: _buildList(_style, _viewModel),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: buildListFooter(_viewModel),
+                )
+              ],
+            )));
+  }
+
+  Widget _buildIndicatorLine(ActionSheetStyle style) {
+    return Center(
+      child: Container(
+        margin: style.indicatorLineEdgePadding,
+        width: style.indicatorLineWidth,
+        height: style.indicatorLineThickness,
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.all(style.indicatorLineRadius),
+          color: style.indicatorLineColor,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildList(ActionSheetStyle style, ActionSheetViewModel viewModel) {
+    return ListView.separated(
+      itemCount: viewModel.actions.length,
+      itemBuilder: (context, index) => ListTile(
+            title: ActionSheetListItem(
+                viewModel: ActionSheetListItemViewModel(
+                    title: viewModel.actions[index].title,
+                    icon: viewModel.actions[index].icon,
+                    type: viewModel.actions[index].type,
+                    checkBoxIcon: viewModel.actions[index].checkBoxIcon,
+                    isSelected: viewModel.actions[index].isSelected,
+                    isDestructive: viewModel.actions[index].isDestructive,
+                    onTap: viewModel.actions[index].onTap)),
+            onTap: () => checkTap(index),
+          ),
+      separatorBuilder: (context, index) => ListDivider.build(),
+    );
+  }
+
+  List<Widget> buildListFooter(ActionSheetViewModel _viewModel) {
+    List<Widget> listBuilder = [
+      RoundedButton.build(
+          RoundedButtonViewModel(title: _viewModel.cancelButtonTitle),
+          style: ActionSheetLargeRoundedButtonStyle())
+    ];
+
+    if (_viewModel.confirmButtonTitle != null) {
+      listBuilder.add(RoundedButton.build(
+          RoundedButtonViewModel(title: _viewModel.confirmButtonTitle),
+          style: ActionSheetLargeRoundedButtonStyle()));
+    }
+
+    return listBuilder;
+  }
+
+  void checkTap(int index) {
+    if (_viewModel.actions[index].type == ActionType.selectable) {
+
+      setState(() {
+        _viewModel.actions[index].isSelected =
+            !_viewModel.actions[index].isSelected;
+
+        if (index == _viewModel.actions.length - 1) {
```

Comment:
????? what's happening here? Is it flipping the state either side of the selected index to clear it?

Reply:
I would expect a function "clearSelection" and "select(index)". each time select(index) is called it will first call "clearSelection"

Reply:
@DavidFD005 can you help here please.

Reply:
Yes I explained to her how to do it

Reply:
I changed it, now is not switching the selection. First it's checking if there is a selection, so if the viewModel gives an item as selected, their checkbox will be marked but if there isn't a selected item, the first one will be selected by default.